### PR TITLE
Fix logout endpoint handling for the My Account pages

### LINKED ISSUE #25288: customer-logout endpoint fails to redirect correctly
Woo's custom customer logout fails to redirect to the specified URL after logout.
Through code review I identified there is both

* an errant approach to specifying and processing post-logout redirects
* errant code which ignores the attempt to specify a redirect and instead always redirects to the page named 'myaccount'

This is related to https://github.com/woocommerce/woocommerce/issues/24243#issuecomment-559162421

## Setup

* Ubuntu 18.04.3
* Apache 2.4.38
* PHP 7.3.12
* Wordpress 5.3.2
* Woocommerce 3.8.1
* Setup Woo using the Wizard with basic -- no addons/plugins, no jetpack, etc.

## Repro

Errant behavior can be seen with similar to the following:

1. Add a filter for `woocommerce_logout_default_redirect_url` and make that filter function always `return home_url();`
2. Visit your website and log into an account
3. Visit the custom Woo "My Account" page
4. Hover over (don't click) Woo's link on the left for logout.

Notice that the URL is based on `/` and appends on it the endpoint for `customer-logout`. For example, without prettylinks, it will look like `https://myserver/?customer-logout&_wpnonce=1234567890`. **_This is the errant approach for redirect support._**

The `home_url()` redirect specified has instead been used as a base URL on which an endpoint was placed. I can think of many scenarios where this will fail to both logout and fail to redirect with URLs to other websites, query params, fragments, ports, etc. 

More repro steps...

1. Click on that logout link

## Result

The user is redirected to the "My Account" page, e.g. without pretty links it is `https://myserver/?page_id=9`

## Expected

Redirect to `home_url()` as was specified. 

## Notes

I don't have an immediate global fix because these errors are both in approach and in specific code. I believe the core team needs to consider how they want to approach redirects on endpoints like logout. With that choice, then specific code can be changed or written.

### Errant approach

The errant approach scatters itself within Woo's code. Here is one example of misusing base URL as a redirect.

https://github.com/woocommerce/woocommerce/blob/aa590c8acd7b2c61327af9abefe446d14125a6fe/includes/wc-template-functions.php#L3486

At that code, the intention is to logout and then redirect to an URL. However, the function used for that intention does not create an URL that later results in a redirect. Instead, the function called creates an URL on which an endpoint is placed on a base URL (worded as a permalink in the params).

https://github.com/woocommerce/woocommerce/blob/aa590c8acd7b2c61327af9abefe446d14125a6fe/includes/wc-page-functions.php#L87

Just a few redirects URLs that will cause failure to logout and/or redirect correctly

```
https://mysameserver:1234/
https://remoteserver/
https://username:password@remoteserver.com/path/something?foobar=67#fragment12
```

### Specific errant hardcoded redirect

There is also a specific code bug that causes Woo to always ignore the desired redirect on logout. Instead, Woo will always redirect the customer to the "My Account" page

https://github.com/woocommerce/woocommerce/blob/aa590c8acd7b2c61327af9abefe446d14125a6fe/includes/wc-template-functions.php#L35

Two problems here

1. The errant approach discussed above is also here.
2. The filter `woocommerce_logout_default_redirect_url` should be applied to the default on that line of code. One place this was correctly done and you can copy the filter is at https://github.com/woocommerce/woocommerce/blob/aa590c8acd7b2c61327af9abefe446d14125a6fe/includes/wc-template-functions.php#L3483

## Isolating the problem (mark completed items with an [x])

- [] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] I have deactivated other plugins **_not needed to reproduce this bug_** and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>

```
### WordPress Environment ###

WordPress address (URL): https://**redacted**
Site address (URL): https://**redacted**
WC Version: 3.8.1
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 150 MB
WP Debug Mode: –
WP Cron: –
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.38
PHP Version: 7.3.12
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.64.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.1.34-MariaDB-1~bionic
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.1
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 1.75MB
Database Data Size: 0.75MB
Database Index Size: 1.00MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.13MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

Secure DB Connection: by Xiao Yu – 1.1.5
** redacted, this plugin adds the code/filter to expose Woo's errant behavior **
WooCommerce: by Automattic – 3.8.1

### Inactive Plugins (19) ###

** redacted **

### Dropin Plugins (1) ###

db.php: Secure DB Connection

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /?page_id=6
Cart: #7 - /?page_id=7
Checkout: #8 - /?page_id=8
My account: #9 - /?page_id=9
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Eleven
Version: 3.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 1
Oldest: 2019-12-20 19:48:47 +0000
Newest: 2019-12-20 19:48:47 +0000

Pending: 0
Oldest: –
Newest: –

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –
```
</details>

### LINKED ISSUE #25433: Logout behaviour via My Account isn't consistent
**Describe the bug**

2647076-zen

Pressing logout via a menu item returns a message - `Are you sure you want to log out? Confirm and log out`.

Pressing the [submenu button `logout`](https://github.com/woocommerce/woocommerce/blob/2e61321e2a6b89d89de92b0270c6cd0e00a44ea9/includes/wc-account-functions.php#L95) or the [`not %1$s? <a href="%2$s">Log out</a>`](https://github.com/woocommerce/woocommerce/blob/2e61321e2a6b89d89de92b0270c6cd0e00a44ea9/templates/myaccount/dashboard.php#L29) immediate logs the customer out. 

Any menu link added via the Customizer has one nonce, and the My Account links have another nonce. 

When the `Are you sure you want to log out? Confirm and log out` is shown, `log out` is now linked to the same nonce as the other two. 

**To Reproduce**
1. Go to [site]/my-account
2. Click on a Logout menu item
3. See that the confirmation message is shown
4. Refresh page
5. Click on the Submenu item or the "not you" link
6. Notice that the customer is logged out

**Screenshots**

woo_logout_nonce

Image link - https://d.pr/i/YNOgbi

**Expected behavior**

Pressing any logout button on the site presents the same behavior - one-click logout. 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only the WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

Introduced in #15571

**WordPress Environment**
<details>
```
### WordPress Environment ###

WordPress address (URL): https://core.doug.press
Site address (URL): https://core.doug.press
WC Version: 3.8.1
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.2.26
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.67.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.21-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.1
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 2.78MB
Database Data Size: 1.62MB
Database Index Size: 1.16MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_advanced_notifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_advanced_notification_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.31MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.27MB + Index: 0.16MB + Engine InnoDB
wp_posts: Data: 0.31MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpml_mails: Data: 0.11MB + Index: 0.00MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Akismet Anti-Spam: by Automattic – 4.1.3
Homepage Control: by WooThemes – 2.0.3
Log Emails: by WebAware – 1.3.1
WooCommerce: by Automattic – 3.8.1

### Inactive Plugins (8) ###

Classic Editor: by WordPress Contributors – 1.5
Gutenberg: by Gutenberg Team – 7.2.0
Jetpack by WordPress.com: by Automattic – 8.1
Storefront Parallax Hero: by WooCommerce – 1.5.7
WooCommerce Blocks: by Automattic – 2.5.10
WooCommerce Royal Mail: by WooCommerce – 2.5.20
WooCommerce Stripe Gateway: by WooCommerce – 4.3.1
WP Mail Logging: by MailPoet – 1.9.5

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.3
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 3
Oldest: 2020-01-04 09:11:13 +0000
Newest: 2020-01-20 11:23:43 +0000

Pending: 0
Oldest: –
Newest: –

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

```
</details>

### PR DESCRIPTION
**Pull request by mikejolley**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes log out link behaviour:

1. Prevents some log out links requiring nonces and some not—all require a nonce
2. Improves navigation menus (in non-block themes) by adding nonce automatically to log out links
3. Adds point 2 to block themes also by using `block_core_navigation_render_inner_blocks`
4. Fixes `wc_logout_url` so it supports redirect regardless of settings (when using the endpoint, the custom redirect would be ignored)
5. Consolidate redirect filters into `wc_get_logout_redirect_url` function
6. Improved `customer-logout` endpoint handling by logging out inline instead of doing another redirect to log out. This also improves performance because it requires less requests.

Closes #25288
Closes #25433

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Log in and go to the "my account" page
2. Confirm the log-out link in the body of the page works and redirects you back to the account page.
3. Log in and go to the "my account" page again
4. Confirm the log out link in the sidebar works and again redirects you to the account page
5. Log in again
6. Navigate to `/my-account/customer-logout/`. Ensure you see a notice asking you to confirm your logout, then click the button in the notice and confirm you're logged out.
7. Using Storefront theme. Add a navigation menu and inside add a "woocommerce endpoint" link for log out. See screenshot below.
8. Log out using this link. It should work and the URL should include a nonce.
9. When logged out, this link should be hidden.

Screenshot 2024-07-17 at 12 45 45

10. Switch to a block theme.
11. Add a navigation block and add a custom link inside for `/my-account/customer-logout/`
12. Log out using this link. It should work and the URL should include a nonce.
13. When logged out, this link should be hidden.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by senadir — APPROVED** [review]
This is testing well with me, it was hard to follow the code due to how complex and nested it gets, but that's not this PR fault.

File: plugins/woocommerce/includes/wc-account-functions.php [hunk 1]
```diff
@@ -176,11 +176,13 @@ function wc_get_account_endpoint_url( $endpoint ) {
 		return wc_get_page_permalink( 'myaccount' );
 	}
 
+	$url = wc_get_endpoint_url( $endpoint, '', wc_get_page_permalink( 'myaccount' ) );
+
 	if ( 'customer-logout' === $endpoint ) {
-		return wc_logout_url();
+		return wp_nonce_url( $url, 'customer-logout' );
 	}
```
Comment (sun) [inline plugins/woocommerce/includes/wc-account-functions.php:183]: ⚠️ This change broke the logout link on existing sites that were using `wc_logout_url()` before to generate the link to the logout page.

Instead of getting logged out, the user is presented with a question to confirm they want to log out.  Clicking the link in the message repeats the same page response, but the user is never logged out.  On top, this is all happening on the `wp-login.php` front controller of WordPress Core instead of the WooCommerce endpoint.

This is because the `customer-logout` endpoint is using a different nonce than the `wp-login.php` front controller.  And because `wc_logout_url()` calls straight into `wp_logout_url()` now, the unexpected endpoint is loading.

Suggestions:
1. This change should have been announced with a much higher severity, because it is not backwards compatible and breaks existing sites.
2. The nonce of the `customer-logout` endpoint could at least have used the same action as the `wp-login.php?action=logout` endpoint; i.e., `'log-out'`, so at least the user's intention is executed, even if on the wrong controller.

```suggestion
	if ( 'customer-logout' === $endpoint ) {
 		return wp_nonce_url( $url, 'log-out' );
	}
```

It was anything but trivial to find the cause, so silence on this PR may not indicate that no one is affected.

Comment (sun) [reply plugins/woocommerce/includes/wc-account-functions.php:183]: For others affected by this change, the remedy is to adjust the logout URL as follows in your plugins and themes:

```diff
-    $context['logout_url'] = wc_logout_url();
+    $context['logout_url'] = wc_get_account_endpoint_url('customer-logout');
```

Comment (senadir) [reply plugins/woocommerce/includes/wc-account-functions.php:183]: I'm not fully up to memory with this PR. @mikejolley would it be possibly to modify `wc_logout_url` so it continues to work?

We might include that in 9.5

Comment (mikejolley) [reply plugins/woocommerce/includes/wc-account-functions.php:183]: @senadir I was aware using the function without nonce would trigger a confirmation notice to confirm logout, but not that there was a problem with the nonces. Let me check.

Comment (mikejolley) [reply plugins/woocommerce/includes/wc-account-functions.php:183]: If you simulate log-out without nonce by navigating to:

```
yourstore.com/my-account/customer-logout
```

You see the prompt/notice (expected). The link in the notice contains a valid logout url with nonce:

```
yourstore.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2Fstore.local%2Fmy-account%2F&_wpnonce=XXX
```

For `wc_logout_url()`, I can see it outputs a valid link with nonce too. This doesn't go via the `customer-logout` endpoint:

```
yourstore.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2Fstore.local%2Fmy-account%2F&_wpnonce=xxx
```

To clarify, links that  go via `yourstore.com/my-account/customer-logout` without nonce now require confirmation.
**This cannot be circumvented because that would allow anyone to trigger logout without using a nonce at all thus rendering all logout nonces pointless.** The confirmation notice is an unfortunate step, but its not breaking.

I also tested `wc_get_account_endpoint_url( 'customer-logout' )` which is not being used any longer in core. This gave me a nonced URL which worked. The handling for that is in here:

https://github.com/woocommerce/woocommerce/blob/05f70533bc6fdb280d7520ccfcb0059128cb74e0/plugins/woocommerce/includes/wc-template-functions.php#L37-L42

```php
if ( ! empty( $_REQUEST['_wpnonce'] ) && wp_verify_nonce( sanitize_key( $_REQUEST['_wpnonce'] ), 'customer-logout' ) ) {
		wp_logout();
		wp_safe_redirect( wc_get_logout_redirect_url() );
		exit;
}
```

This seems fine to me. No bug here @senadir
